### PR TITLE
Removes REST and ElasticSearch

### PR DESCRIPTION
**Pull request by JPrevost**
Why are these changes being introduced:

* We have moved to OpenSearch and GraphQL

Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/TIMX-220
* https://mitlibraries.atlassian.net/browse/TIMX-213

How does this address that need:

* Removes v2 feature flag and all code in the "else" blocks of v2 checks
* Removes ElasticSearch code
* Removes no longer used dependencies
* Removes most of the REST API. The auth portion will remain until the user model is removed in https://mitlibraries.atlassian.net/browse/TIMX-221
* All unused cassettes were removed (this was tedious)
* Updates Actions workflow to no longer run the tests twice
* Updates documentation (public and dev facing)

Document any side effects to this change:

* The auth portion of the REST API remains as adding removal of the user model and associated code was too much scope creep. An additional ticket has been added to remove the User model and auth portion of the REST API.
* The dockerfile is still very elasticsearch focused. There is a separate ticket to address this so I chose to leave it unchanged in this commit.
* There is a separte ticket to move to our shared Actions CI workflow

#### Developer

- [x] All new ENV is documented in README
- [ ] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/views/static/index.html.erb [hunk 1]
```diff
@@ -33,27 +33,21 @@
...
-      <div class="panel panel-info">
-        <div class="panel-heading">Quick Example</div>
-        <div class="panel-body">
-          <p>You can do a very quick test of your account via this `curl` example. The documentation portal linked above has much more complete information.</p>
-
-          <code>curl -X GET --header 'Authorization: Bearer <%= JwtWrapper.encode({ user_id: current_user.id }) %>' '<%= api_v1_search_url(q: 'whatever')%>'</code>
-        </div>
-      </div>
-    <% end %>
+      <p>Our original REST API was deprecated and then retired in early 2003. The replacement GraphQL API can search more data and has features not found in the origial API. If you need help migrating from REST to GraphQL please <a href="mailto:timdex@mit.edu?Subject=TIMDEX%20GraphQL%20transition">let us know</a>.</p>
```
Comment (jazairi) [inline app/views/static/index.html.erb:49]: If it was retired in early 2003, then the time distortion I've been experience recently is more dramatic than I wagered. 🙂

File: README.md [hunk 2]
```diff
@@ -79,46 +56,30 @@ locally.
   `yourapp.herokuapp.com`. However, if you use a custom domain in production,
   that should be the value you use in production.
 - `JWT_SECRET_KEY`: generate with `rails secret`
-- `ELASTICSEARCH_INDEX`: Elasticsearch index or alias to query
-- `ELASTICSEARCH_URL`: defaults to `http://localhost:9200`
+- `OPENSEARCH_INDEX`: Opensearch index or alias to query
+- `OPENSEARCH_URL`: Opensearch URL
```
Comment (jazairi) [inline README.md:60]: I'm able to execute queries locally without a defined `OPENSEARCH_INDEX`, so I'm not sure that's required in all environments. As for `OPENSEARCH_URL`, if we are no longer defaulting to `localhost:9200`, I think we should add it to`.env.test`.

**Review by jazairi — COMMENTED** [review]
Great work on this! See my comments about the readme changes, particularly around `OPENSEARCH_INDEX` and `OPENSEARCH_URL`. They are minor concerns, but I think it's important to make sure we have the appropriate context in our docs.

File: app/views/static/index.html.erb (hunk 1 above)
Comment (JPrevost) [reply app/views/static/index.html.erb:49]: I genuinely have a harder time with years than months or days... but also, will fix :)

File: README.md (hunk 2 above)
Comment (JPrevost) [reply README.md:60]: Interesting. I'm curious what is being searched if not index is provided. I'll confirm that behavior and check whether these are required or optional. 👍🏻

Comment (JPrevost) [reply README.md]: When no index is provided, it searches _all_ indexes. This is likely not a great idea so I updated the readme to suggest the more useful aliases.

I also moved these two to required in prod... which hopefully is interpreted as optional elsewhere?

Comment (jazairi) [reply README.md]: Thanks for adding that context. Yeah, I think it's an established pattern in our docs that specifying something as required in one environment implicitly specifies it as optional elsewhere.

**Review by jazairi — APPROVED** [review]

**Pull request merged**